code/game/objects/items/charter.dm

```diff
@@ -121,3 +121,37 @@
 		used = TRUE
 
 #undef STATION_RENAME_TIME_LIMIT
+
+/obj/item/station_charter/revolution
+	name = "revolutionary banner"
+	desc = "A banner symbolizing a bloody victory over treacherous tyrants."
+	icon = 'icons/obj/banner.dmi'
+	icon_state = "banner_revolution"
+	inhand_icon_state = "banner-red"
+	lefthand_file = 'icons/mob/inhands/equipment/banners_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/banners_righthand.dmi'
+	w_class = 5
+	force = 15
+	ignores_timeout = TRUE //non roundstart!
+	var/timesup = FALSE
+
+/obj/item/station_charter/revolution/Initialize()
+	. = ..()
+	addtimer(VARSET_CALLBACK(src, timesup, TRUE), 5 MINUTES)
+
+/obj/item/station_charter/revolution/attack_self(mob/living/user)
+	if(timesup && !used)
+		to_chat(user, "<span class='warning'>You have lost the victorious fervor to declare a new name.</span>")
+		return
+	. = ..()
+
+/obj/item/station_charter/revolution/rename_station(designation, uname, ureal_name, ukey)
+	. = ..()
```

Comment:
cut this 💅

Reply:
is this not the place admins get allowed to veto names and such?

Reply:
This is after a name is accepted, it's calling parent which is also setting the name and announcing it. Eek!